lib/inspec/resources/sys_info.rb

```diff
@@ -13,13 +13,36 @@ class System < Inspec.resource(1)
       describe sys_info do
         its('hostname') { should eq 'example.com' }
       end
+
+      describe sys_info do
+        its('fqdn') { should eq 'user.example.com' }
+      end
+
     EXAMPLE
 
+    %w{ domain fqdn ip-address short }.each do |opt|
+      define_method(opt.to_sym) do
+        hostname(opt)
+      end
+    end
+
     # returns the hostname of the local system
-    def hostname
+    def hostname(opt = nil)
       os = inspec.os
       if os.linux? || os.darwin?
```

Comment:
Hrm, this is not portable to Darwin (MacOS):
```
hostname: illegal option -- I
usage: hostname [-fs] [name-of-host]
```

Neither -d nor -I works there - I think we either need a no-support exception, or an alternate implementation.

Reply:
I don't think I have access to a mac right now at least not that I can get setup with inspec. From the man pages though it looks like it can only do full or trim the domain. I'm not sure an alternate implementation is possible but I'm not a mac user.

Reply:
I refactored it so it should be portable on MacOS now. I couldn't test it but I based it off the mac hostname man page so I was able to still implement a couple of the options. I also added failures for unsupported options for all platforms.